src/main/java/com/google/cloud/tools/maven/CloudSdkAppEngineFactory.java

```diff
@@ -154,8 +159,23 @@ private AppEngineDevServer createDevServerForVersion(
    */
   public AppEngineDevServer devServerRunAsync(
       int startSuccessTimeout, SupportedDevServerVersion version) {
+    File logDir =
+        Paths.get(mojo.mavenProject.getBuild().getDirectory())
+            .resolve("dev-appserver-out")
+            .toFile();
+    if (!logDir.exists() && !logDir.mkdirs()) {
```

Comment:
nit: since this is all new code, we generally prefer to go with the `nio.Path` API rather than the ([kinda deprecated](https://stackoverflow.com/questions/30547589/why-isnt-java-io-file-deprecated)) `io.File`. For example, since you are working with a Path already, the exists and mkdirs can be done with `Files.exists` and `Files.createDirectories`.